api/providers/ClientProvider.cs

```diff
@@ -0,0 +1,71 @@
+using DTO.Client;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+
+public class ClientProvider : BaseProvider<Client>
+{
+    private readonly IValidator<Client> _clientValidator;
+    private readonly ContactProvider _contactProvider;
+    private readonly AddressProvider _addressProvider;
+
+    public ClientProvider(AppDbContext db, IValidator<Client> clientValidator, ContactProvider contactProvider, AddressProvider addressProvider)
+        : base(db)
+    {
+        _clientValidator = clientValidator;
+        _contactProvider = contactProvider;
+        _addressProvider = addressProvider;
+    }
+
+
+    public override Client? Create(BaseDTO createValues)
+    {
+        ClientRequest request = createValues as ClientRequest ?? throw new ApiFlowException("Could not process create client request. Save new client failed.");
+        
+        if (request.ContactId == null && request.Contact == null) throw new ApiFlowException("Either contact_id or contact fields must be filled.");
+        if (request.AddressId == null && request.Address == null) throw new ApiFlowException("Either address_id or address fields must be filled.");
+
+        Contact? relatedContact = GetOrCreateContact(request);
+        Address? relatedAddress = GetOrCreateAddress(request);
+        
+        if (relatedContact == null || relatedAddress == null) throw new ApiFlowException("Failed to process contact or address.");
+        
+        Client newClient = new Client(newInstance: true)
+        {
+            Name = request.Name,
+            ContactId = relatedContact.Id,
+            AddressId = relatedAddress?.Id
+        };
+
+        ValidateModel(newClient);
```

Comment:
```suggestion
                Client newClient = new Client(newInstance: true)
        {
            Name = request.Name,
        };

        if(relatedContact != null) newClient.ContactId = relatedContact.Id;
        if(relatedContact != null) newClient.AddressId = relatedAddress.Id;

        ValidateModel(newClient);
```

Ik zou het op deze manier maken want pas als de relatedAddress of Contact niet leeg is dan pas vul je hem in bij de newClient. Als 1 van de condities niet true zijn dan pak de validateModel functie het op en geeft een mooie response